File: daily/面试题 02.02. 返回倒数第 k 个节点.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {number}
 */
var kthToLast = function(head, k) {
    var fast=head,slow=head,flag=0;
    while(fast)
    {
        if(flag>=k)
            slow=slow.next;
        fast=fast.next;
        flag++;
    }
    return slow.val;
};

File: daily/C++版本/面试题 02.02. 返回倒数第 k 个节点.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    int kthToLast(ListNode* head, int k) {
        ListNode *fast=head,*slow=head;
        int flag=0;
        while(fast)
        {
            if(flag>=k)
                slow=slow->next;
            fast=fast->next;
            flag++;
        }
        return slow->val;
    }
};
